src/selfplay/game.cc

```diff
@@ -91,19 +97,39 @@ SelfPlayGame::SelfPlayGame(PlayerOptions white, PlayerOptions black,
     tree_[1] = std::make_shared<NodeTree>();
     tree_[1]->ResetToPosition(orig_fen_, {});
   }
+  int ply = 0;
+  auto white_prob = white.uci_options->Get<float>(kOpeningStopProbId);
+  auto black_prob = black.uci_options->Get<float>(kOpeningStopProbId);
+  if (white_prob != black_prob && white_prob * black_prob > 0.0f) {
+    throw Exception("Stop probabilities must be both equal or zero!");
+  }
+
   for (Move m : opening.moves) {
+    auto exit_prob_now = tree_[0]->IsBlackToMove() ? black_prob : white_prob;
+    auto exit_prob_next = tree_[0]->IsBlackToMove() ? white_prob : black_prob;
+    int rem_moves = opening.moves.size() - ply;
```

Comment:
If we are looking at the last move of the opening book (or a line consisting of a single move), we have 2 exit options left to decide between, but I believe this returns 1. We can either manually add +1 to `rem_moves` below, or change `rem_moves` into `rem_positions` and add the +1 at here.